Lib/asyncio/base_events.py

```diff
@@ -295,8 +296,10 @@ def _attach(self, transport):
         self._clients.add(transport)
 
     def _detach(self, transport):
-        assert transport in self._clients
-        self._clients.remove(transport)
+        # Note that 'transport' may already be missing from
+        # self._clients if it has been garbage collected
+        if transport in self._clients:
+            self._clients.remove(transport)
```

Comment:
There's a simpler way: use `discard()` instead of `remove()`.
```suggestion
        self._clients.discard(transport)
```